### PR TITLE
ref: Normalize user options

### PR DESCRIPTION
**Pull request by lforst**
This PR adds normalization of user options that should make it easier to work with them down the line.

### CODE REVIEW

**Comment by lforst** [comment]
> At some point, I'd like to see us having tests for the options normalization but it doesn't have to be right now. Let's talk about how we want to deal with tests for now and go with that strategy.

Screw tests.

Jokes aside: Probably yes. Here it doesn't matter too much imo because TS pretty much asserts that stuff is correct but at some point, we should probably do some high-level integration tests.

File: packages/bundler-plugin-core/src/index.ts [hunk 1]
```diff
@@ -211,15 +200,15 @@ const unplugin = createUnplugin<Options>((originalOptions, unpluginMetaContext)
...
+        if (typeof internalOptions.entries === "function") {
+          return internalOptions.entries(id);
         }
 
-        const arrayifiedEntriesOption = Array.isArray(options.entries)
-          ? options.entries
-          : [options.entries];
+        const arrayifiedEntriesOption = Array.isArray(internalOptions.entries)
+          ? internalOptions.entries
+          : [internalOptions.entries];
```
[the remark this answers is not in the document]
Comment (lforst) [reply packages/bundler-plugin-core/src/index.ts]: Yeah thanks. Didn't notice that! f7ccddccc6bbbe467e8131c0fa85b047cec04564

File: packages/bundler-plugin-core/src/index.ts [hunk 2]
```diff
@@ -297,11 +284,11 @@ const unplugin = createUnplugin<Options>((originalOptions, unpluginMetaContext)
...
 
-      createNewRelease(release, options, ctx)
-        .then(() => cleanArtifacts(release, options, ctx))
-        .then(() => uploadSourceMaps(release, options, ctx))
+      createNewRelease(internalOptions.release, internalOptions, ctx)
+        .then(() => cleanArtifacts(internalOptions.release, internalOptions, ctx))
+        .then(() => uploadSourceMaps(internalOptions.release, internalOptions, ctx))
         .then(() => setCommits(ctx)) // this is a noop for now
-        .then(() => finalizeRelease(release, options, ctx))
+        .then(() => finalizeRelease(internalOptions.release, internalOptions, ctx))
```
[the remark this answers is not in the document]
Comment (lforst) [reply packages/bundler-plugin-core/src/index.ts]: Totally! f7ccddccc6bbbe467e8131c0fa85b047cec04564

File: packages/bundler-plugin-core/src/options-mapping.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        return convertIncludePathToIncludeEntry(potentialIncludeEntry);
+      } else {
+        return potentialIncludeEntry;
+      }
+    });
+  } else {
+    userInclude = [userOptions.include];
+  }
+
+  const include = userInclude.map((userIncludeEntry) => normalizeIncludeEntry(userIncludeEntry));
```
[the remark this answers is not in the document]
Comment (lforst) [reply packages/bundler-plugin-core/src/options-mapping.ts]: Yeah forgot about the conversation haha. Thanks for reminding me! 5edfdf9370ecdd1608a3d43b9ae2e0f4001867ea

**Review by Lms24 — APPROVED** [review]
Nice!

**Pull request merged**